tests/failing/js-args.ml

```diff
@@ -118,13 +118,18 @@ let _ =
       <= b)
   then ()
 let _ =
+  (* Comparisons are different than conditionals; we don't regard them as
+     conceptually part of the [if] expression. *)
   if a
      <= b
   then ()
 let _ =
+  (* We regard the outermost condition terms as conceptually part of the [if]
+     expression and indent accordingly.  Whether [&&] or [||], conditionals
+     effectively state lists of conditions for [then]. *)
   if Edge_adjustment.is_zero arb.cfg.extra_edge
-     && 0. = sys.plugs.edge_backoff
-     && 0. = zero_acvol_edge_backoff
+  && 0. = sys.plugs.edge_backoff
+  && 0. = zero_acvol_edge_backoff
   then 0.
   else 1.
 let _ =
```

Comment:
On Mon, Jul 27, 2015 at 9:04 PM, Louis Gesbert <notifications@github.com>
wrote:

> In tests/failing/js-args.ml
> <https://github.com/OCamlPro/ocp-indent/pull/185#discussion_r35605876>:
>
> >    if Edge_adjustment.is_zero arb.cfg.extra_edge
> > -     && 0. = sys.plugs.edge_backoff
> > -     && 0. = zero_acvol_edge_backoff
> > +  && 0. = sys.plugs.edge_backoff
> > +  && 0. = zero_acvol_edge_backoff
> >    then 0.
> >    else 1.
> >  let _ =
>
> Should it be the same for when ? E.g.
>
> let f = function
>   | x when c1
>         && c2 ->
>     x
>
> I think you have discretion in that case.  I can ask around but that's
just begging for trouble.  Obivously, we may change our mind in the future,
but I haven't heard any opinions about "when".

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OCamlPro/ocp-indent/pull/185/files#r35605876>.
>